File: ClientApp/src/app/clases/antecedentetitulo.ts
export class antecedentetitulo{
    id : number;
    relaciondocencia : string; //es "byte" en c# y en la Db es "tinyint"
    lugar : string;
    fechainicio : Date;
    fechafin : Date;
    puntajedocente : number;
    descripcion : string;
    titulo : string; 
    tipotitulo: string;
    idcurriculum:number;

    
    
    constructor(id,relaciondocente,lugar,fechainicio,fechafin,puntajedocente,descripcion,titulo,tipotitulo,idcurriculum){
        //el orden de os parametros afecta la vista de los input en el modal
        this.id = +id;
        this.relaciondocencia = relaciondocente;
        this.lugar =  lugar;
        this.fechainicio = new Date (fechainicio);
        this.fechafin = new Date(fechafin);
        this.puntajedocente= +puntajedocente;
        this.descripcion =  descripcion;
        this.titulo =  titulo; 
        this.tipotitulo= tipotitulo;
        this.idcurriculum= +idcurriculum;
         // formateo a number el string // parseInt(val

    }
    
}

File: modelo/antecedentetitulo.cs
using System.Data;
using System;

public class antecedentetitulo : oObjeto{

public string Relaciondocencia{get; set;}
public string Lugar {set;get;}
public DateTime Fechainicio{set;get;}
public DateTime Fechafin{set;get;}
public int Puntajedocente{get;set;}
public string Descripcion{get;set;}
public string Titulo{get;set;}
public string Tipotitulo{set;get;}
public int Idcurriculum { get; set; }

public antecedentetitulo(){}

public antecedentetitulo(DataRow dr)
{
this.Id=Convert.ToInt32(dr["id"]);
this.Relaciondocencia= Convert.ToString(dr["relaciondocencia"]);
this.Lugar= Convert.ToString(dr["lugar"]);
this.Fechainicio= Convert.ToDateTime(dr["fechainicio"]);
this.Fechafin=Convert.ToDateTime(dr["fechafin"]);
this.Puntajedocente= Convert.ToInt32(dr["puntajedocente"]);
this.Descripcion= dr["descripcion"].ToString();
this.Titulo= dr["titulo"].ToString();
this.Tipotitulo=Convert.ToString(dr["tipotitulo"]);
this.Idcurriculum=Convert.ToInt32(dr["idcurriculum"]);
}

}